### ISSUE TITLE
Code Structure and Organization

### ISSUE THREAD

**Issue by MadhavM-117** [MadhavM-117/pacmanJS#6]
Want to be on the same page, regarding code structure and organization. 

- Let's keep the top level directory, clean. 
- All typescript/javascript should be in the `src` directory. 
- organize files into an appropriate folder structure, where possible. A flat structure will get too big, and overwhelming, quite quickly.
- avoid files getting too large. 
- create classes where possible. 
- avoid using the `any` type. 
- be explicit with types, where possible

**Comment by adhicoder** [comment]
Cool, let's also do a code review when every pull request is merged. That
way we're on the same page and also keeping a check.

**Comment by MadhavM-117** [comment]
Are you suggesting we only merge to master, using PRs, and an approval from the other person? I'm okay with that approach.

**Comment by MadhavM-117** [comment]
I can set up branch protection to support that workflow.

**Comment by adhicoder** [comment]
Yes, that's what I meant. Merge should be approved by another person after review.
